thrall/app/lib/elasticsearch/EsInfo.scala

```diff
@@ -0,0 +1,17 @@
+package lib.elasticsearch
+
+import play.api.libs.json.Json
+
+case class MigrationEsInfo(
+  migratedTo: List[String] = Nil,
+  failures: List[String] = Nil,
+)
+object MigrationEsInfo {
+  implicit val format = Json.format[MigrationEsInfo]
+}
+
+case class EsInfo(migration: Option[MigrationEsInfo] = None)
```

Comment:
TODO: try restructuring this: maybe using private/public apply methods to ensure only one of migratedTo or failures is set, or perhaps using an Either

Reply:
done using an Either